cpp/src/arrow/compute/api_vector.cc

```diff
@@ -140,6 +144,15 @@ PartitionNthOptions::PartitionNthOptions(int64_t pivot)
     : FunctionOptions(internal::kPartitionNthOptionsType), pivot(pivot) {}
 constexpr char PartitionNthOptions::kTypeName[];
 
+SelectKOptions::SelectKOptions(int64_t k, std::vector<std::string> keys, std::string keep,
+                               SortOrder order)
+    : FunctionOptions(internal::kSelectKOptionsType),
+      k(k),
+      keys(std::move(keys)),
+      keep(keep),
+      order(order) {}
+constexpr char SelectKOptions::kTypeName[];
+
```

Comment:
The select K algorithm is a general approach to get the topK, bottomK, or median statistic. It seems that SortOrder option is always `Descending` for topK and `Ascending` for bottomK, so I recommend to use an enum for the type of statistic desired instead of specifying ordering. As a user, if I specify `Ascending` it is not intuitive that it corresponds to topK because it depends from which side the sorted data is searched.
```
enum class SelectionOperator {
  TOPK,
  BOTTOMK,
  MEDIAN,  // possibly for another PR
};
```
Also, I am not sure that the`keys` and `keep` options are part of common SelectK algorithms.

Reply:
`keep` comes from Pandas [nlargest/nsmallest](https://pandas.pydata.org/docs/reference/api/pandas.core.groupby.SeriesGroupBy.nlargest.html) however it is not meaningful here unless there is an ordering defined on the dataset.

`keys` is because you may wish to sort on one column but take values from another. However, as mentioned below, if indices were returned instead of values, you could do this yourself instead of having to build it into the kernel.

Reply:
Ok, got it. Thanks for the clarification.